### PR TITLE
Fetch attestation endorsement

### PR DESCRIPTION
**Pull request by takuro-sato**
A part of https://github.com/microsoft/CCF/issues/4667
- Fetch attestation endorsement
- Rename collateral to endorsement for consistency

### CODE REVIEW

File: attestation-container/protobuf/attestation-container.proto [hunk 1]
```diff
@@ -6,7 +6,7 @@ package attestation_container;
 
 // attestation_container service definition.
 service AttestationContainer {
-  // Fetchs and returns attestation report and collateral.
+  // Fetches and returns attestation report and its endorsement.
```
Comment (jumaffre) [inline attestation-container/protobuf/attestation-container.proto:9]:
```suggestion
  // Fetches and returns attestation report and its endorsements.
```

File: attestation-container/protobuf/attestation-container.proto [hunk 2]
```diff
@@ -17,4 +17,5 @@ message FetchAttestationRequest {
 
 message FetchAttestationReply {
   bytes attestation = 1;
+  bytes attestation_endorsement = 2;
```
Comment (jumaffre) [inline attestation-container/protobuf/attestation-container.proto:20]: I think we should call this `attestation_endorsement_certificates` or similar, as we'll also add the certificate revocation list too, once supported by AMD?

File: attestation-container/attest/attestation_endorsement.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		}
+	}
+	return nil, err
+}
+
+func fetchAttestationEndorsementAzure(reportedTCBBytes [REPORTED_TCB_SIZE]byte, chipID string) ([]byte, error) {
+	// Fetch attestation endorsement from Azure endpoint
+	reportedTCB := binary.LittleEndian.Uint64(reportedTCBBytes[:])
+	reportedTCBHex := fmt.Sprintf("%x", reportedTCB)
+	requestURL := fmt.Sprintf("https://global.acccache.azure.net/SevSnpVM/certificates/%s/%s?api-version=2020-10-15-preview", chipID, reportedTCBHex)
```
Comment (jumaffre) [inline attestation-container/attest/attestation_endorsement.go:68]: I feel like the Azure and AMD endpoint addresses should be moved to the top of the file as global constants?

**Review by jumaffre — APPROVED** [review]

File: attestation-container/protobuf/attestation-container.proto (hunk 2 above)
Comment (takuro-sato) [reply attestation-container/protobuf/attestation-container.proto:20]: It makes sense to me

File: attestation-container/attest/attestation_endorsement.go (hunk 3 above)
Comment (takuro-sato) [reply attestation-container/attest/attestation_endorsement.go]: [51b028d](https://github.com/microsoft/CCF/pull/4795/commits/51b028d94292869e9f69a8d82b354b781be01785)

File: attestation-container/protobuf/attestation-container.proto (hunk 2 above)
Comment (takuro-sato) [reply attestation-container/protobuf/attestation-container.proto]: [cd1c26d](https://github.com/microsoft/CCF/pull/4795/commits/cd1c26d0977ac6f14b808b0960d4271511cab7a8)

**Pull request merged**